### PR TITLE
.travis.yml: Escape script after unsupported check

### PR DESCRIPTION
**Pull request by jayvdb**
Add an ``exit`` if UNSUPPORTED==true, and move the other
commands up to the top level of script block.

Follows style of https://github.com/coala/coala-bears/commit/a0f258e

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -62,24 +60,24 @@ before_script:
 
 script:
   - >
-    if [[ $TRAVIS_PYTHON_VERSION_MINOR < 4 || $TRAVIS_PYTHON_VERSION_MAJOR == 2 ]]; then
+    if [[ "$UNSUPPORTED" == true ]]; then
       coverage run setup.py install | grep -q 'coala supports only python 3.4 or later'
+    fi
+  - if [[ "$UNSUPPORTED" == true ]]; then exit; fi
```
Comment (RohanVB) [inline .travis.yml]:
```
if [[ "$UNSUPPORTED" == true ]]; then 
exit; 
fi
```
or just the `fi` in next line?

File: .travis.yml [hunk 2]
```diff
@@ -62,24 +60,24 @@ before_script:
...
+  - bash .misc/tests.sh
+  - python setup.py bdist_wheel
+  - pip install ./dist/coala-*.whl
+  - pip install coala-bears[alldeps] --pre -U
+  # https://github.com/coala/coala-bears/issues/1037
```
Comment (RohanVB) [inline .travis.yml]: Why is this here?

File: .travis.yml (hunk 1 above)
Comment (jayvdb) [inline .travis.yml]: why?  Travis shows only the first line, so ideally the `exit` is there so the reader can see what is being done.
Why hide the `if`?

File: .travis.yml (hunk 2 above)
Comment (jayvdb) [inline .travis.yml]: This is an indentation only change, and should not be changed in this commit.
You can read more about it at https://github.com/coala/coala-bears/issues/1037

File: .travis.yml (hunk 1 above)
Comment (RohanVB) [inline .travis.yml]: IMO on the next line because it just made more sense in my mind as a block for readability.

File: .travis.yml [hunk 3]
```diff
@@ -62,24 +60,24 @@ before_script:
...
+  - >
+    if [[ "$TRAVIS_PULL_REQUEST" == "false" ]]; then
+      python3 -m nltk.downloader punkt maxent_treebank_pos_tagger averaged_perceptron_tagger
```
Comment (RohanVB) [inline .travis.yml]: Can we load from cache here? We have `pip install coala-bears[alldeps] --pre -U` with the nltk merge

Comment (jayvdb) [inline .travis.yml]: Waiting for https://github.com/coala/coala/pull/4128

File: .travis.yml [hunk 4]
```diff
@@ -62,22 +60,22 @@ before_script:
...
-      pip install coala-bears[alldeps] --pre -U
-      # https://github.com/coala/coala-bears/issues/1037
-      if [[ "$TRAVIS_PULL_REQUEST" != "false" ]]; then
-        sed -i.bak '/bears = GitCommitBear/d' .coafile
-      fi
-      coala --non-interactive
-      make -C docs clean
-      python setup.py docs
     fi
+  - if [[ "$UNSUPPORTED" == true ]]; then codecov || true; exit; fi
```
Comment (Mixih) [inline .travis.yml]: why don't we add this to the if block above?

Comment (jayvdb) [inline .travis.yml]: because then we need `set -x -e` , and need to add trailing `;`, etc

File: .travis.yml [hunk 5]
```diff
@@ -62,22 +60,22 @@ before_script:
...
+  - bash .misc/tests.sh
```
Comment (Mixih) [inline .travis.yml]: intentional omission of the `set -e`?

Comment (jayvdb) [inline .travis.yml]: Yes, travis does `set -e` by checking the error status of each entry in the build script.  It needs to be manually added only if there are multiple bash commands in a single entry of the build script, which travis cant 'see'.

**Comment by Mixih** [comment]
ack 3b258e7

**Comment by jayvdb** [comment]
@rultor merge
